watertap/unit_models/boron_removal.py

```diff
@@ -0,0 +1,1123 @@
+            doc="Charge of the caustic additive",
+        )
+        self.caustic_dose = Var(
+            self.flowsheet().config.time,
+            initialize=0,
+            bounds=(0, None),
+            domain=NonNegativeReals,
+            units=pyunits.mg / pyunits.L,
+            doc="Dosages of the set of caustic additive",
+        )
+
+        # Reaction parameters
+        self.Kw_0 = Param(
+            mutable=True,
+            initialize=60.91,
+            domain=NonNegativeReals,
+            units=pyunits.mol**2 / pyunits.m**6,
+            doc="Water dissociation pre-exponential constant",
+        )
+        self.dH_w = Param(
+            mutable=True,
+            initialize=55830,
+            domain=NonNegativeReals,
+            units=pyunits.J / pyunits.mol,
+            doc="Water dissociation enthalpy",
+        )
+        self.Ka_0 = Param(
+            mutable=True,
+            initialize=0.000163,
+            domain=NonNegativeReals,
+            units=pyunits.mol / pyunits.m**3,
+            doc="Boron dissociation pre-exponential constant",
+        )
+        self.dH_a = Param(
+            mutable=True,
+            initialize=13830,
+            domain=NonNegativeReals,
+            units=pyunits.J / pyunits.mol,
+            doc="Boron dissociation enthalpy",
+        )
+
+        # molarity vars (for approximate boron speciation)
+        #       Used to establish the mass transfer rates by
+        #       first solving a coupled equilibrium system
+        #
+        #   ENE: [H+] = [OH-] + [A-] + (Alk - n*[base])
+        #   MB:  TB = [HA] + [A-]
+        #   rw:  Kw = [H+][OH-]
+        #   ra:  Ka[HA] = [H+][A-]
+        #
+        #       Alk = sum(n*Anions) - sum(n*Cations) (from props)
+        #       [base] = (Dose/MW)
+        #       TB = [HA]_inlet + [A-]_inlet (from props)
+
+        self.mol_H = Var(
+            self.flowsheet().config.time,
+            initialize=1e-4,
+            bounds=(0, None),
+            domain=NonNegativeReals,
+            units=pyunits.mol / pyunits.m**3,
+            doc="Resulting molarity of protons",
+        )
+        self.mol_OH = Var(
+            self.flowsheet().config.time,
+            initialize=1e-4,
+            bounds=(0, None),
+            domain=NonNegativeReals,
+            units=pyunits.mol / pyunits.m**3,
+            doc="Resulting molarity of hydroxide",
+        )
+        self.mol_Boron = Var(
+            self.flowsheet().config.time,
+            initialize=1e-2,
+            bounds=(0, None),
+            domain=NonNegativeReals,
+            units=pyunits.mol / pyunits.m**3,
+            doc="Resulting molarity of Boron",
+        )
+        self.mol_Borate = Var(
+            self.flowsheet().config.time,
+            initialize=1e-2,
+            bounds=(0, None),
+            domain=NonNegativeReals,
+            units=pyunits.mol / pyunits.m**3,
+            doc="Resulting molarity of Borate",
+        )
+

```

Comment:
So, aren't these the molarities at the outlet? Is there a reason why these couldn't be represented by `self.properties_out[t].conc_mol_phase_comp`?

Reply:
I'm all about approaches that make things more helpful although this might deviate from our conventions. I don't feel strongly about changing what you have necessarily, but we probably want @andrewlee94 and @TimBartholomew 's opinions on this too.

Reply:
Will update names to standard naming conventions to make this more clear.